src/mds/Server.cc

```diff
@@ -5546,6 +5546,19 @@ void Server::handle_client_setdirlayout(MDRequestRef& mdr)
 }
 
 // XATTRS
+int Server::check_if_pool_is_empty(const OSDMap& osdmap, int64_t pool_id)
+{
+  /* Since we don't allow to create pool level snaps for all pools associated
+   * with a CephFS, we also block a pool with pre-existing snaps to be attached
+   * to a CephFS.
```

Comment:
Does that mean we want to prevent a pool that has pre-existing snaps to be used as a FS data pool using 'add_data_pool' or 'fs new' command?  So we will want this check in "FileSystemCommandHandler::_check_pool() as well?

Reply:
Yes; we'd need a check there as well. But I want others to comment on this specific issue.

Reply:
I think that would be an ideal thing to do.